hfile.c

```diff
@@ -821,6 +862,41 @@ static int init_add_plugin(void *obj, int (*init)(struct hFILE_plugin *),
     return 0;
 }
 
+hFILE *hopenv_mem(const char *filename, const char *mode, va_list args)
```

Comment:
I'm wondering if the existing hopen_mem, which is strictly URL based rather than generic memory based, should be reimplemented in terms of the generic hopenv_mem.

Reply:
Looks like a good idea, though in `hopenv_mem` there isn't (and I don't think it would be a good idea to) a way to specify a different buffer size from the length of the buffer that is filled - this is needed in `hopen_mem`. I think it would be a good to create a function `create_hfile_mem`, which would do the functionality in common with both functions (which I have implemented below)